store/s3storeKMS.go

```diff
@@ -0,0 +1,486 @@
+package store
```

Comment:
A bunch of this code is duplicated from the S3 store (obviously :P). I wonder if we could cut down on the duplication by embedding an S3Store in S3KMSStore and overriding functions where there's a difference? Hopefully with 0 refactoring of S3Store

https://medium.com/random-go-tips/method-overriding-680cfd51ce40 for some tips

Reply:
I was thinking the same thing, I just wasn't aware of how to override in `go`. Thanks for the tip!